tests/test_dataset_common.py

```diff
@@ -75,6 +76,27 @@ def wrapper(self, dataset_name):
         return test_case
 
 
+@contextmanager
+def prepare_only_one_shard(dataset_builder: DatasetBuilder):
```

Comment:
I added this function in the dummy data tests to not have to have one dummy data file per shard of the dataset, but only require one shard for the dummy data.